crypto/x509/v3_utf8.c

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2020 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdio.h>
+#include "internal/cryptlib.h"
+#include <openssl/asn1.h>
+#include <openssl/conf.h>
+#include <openssl/x509v3.h>
+#include "ext_dat.h"
+
+const X509V3_EXT_METHOD v3_utf8_list[3] = {
+    EXT_UTF8STRING(NID_subjectSignTool),
+    EXT_UTF8STRING(NID_issuerSignTool),
+    EXT_END
+};
+
+char *i2s_ASN1_UTF8STRING(X509V3_EXT_METHOD *method,
+                          ASN1_UTF8STRING *utf8)
+{
+    char *tmp;
+
+    if (utf8 == NULL || utf8->length == 0) {
+        return NULL;
+    }
+    if ((tmp = OPENSSL_malloc(utf8->length + 1)) == NULL) {
```

Comment:
use OPENSSL_zalloc and get rid of explicit setting last byte to zero?

Reply:
fixed